cluster-health-check/cluster-health.sh

```diff
@@ -0,0 +1,519 @@
+  ${KUBECTL_CMD} get nodes --selector='node-role.kubernetes.io/etcd' -o wide > $TMPDIR/nodes/get-nodes-etcd.wide 2>&1
+  NumberOfEtcdNodes=`${KUBECTL_CMD} get nodes --selector='node-role.kubernetes.io/etcd' --no-headers 2>/dev/null | wc -l`
+  decho "controlplane"
+  ${KUBECTL_CMD} get nodes --selector='node-role.kubernetes.io/controlplane' -o wide > $TMPDIR/nodes/get-nodes-controlplane.wide 2>&1
+  NumberOfControlplaneNodes=`${KUBECTL_CMD} get nodes --selector='node-role.kubernetes.io/controlplane' --no-headers 2>/dev/null | wc -l`
+  decho "worker"
+  ${KUBECTL_CMD} get nodes --selector='node-role.kubernetes.io/worker' -o wide > $TMPDIR/nodes/get-nodes-worker.wide 2>&1
+  NumberOfWorkerNodes=`${KUBECTL_CMD} get nodes --selector='node-role.kubernetes.io/worker' --no-headers 2>/dev/null | wc -l`
+  techo "Node Summary Report"
+  techo "Total number of nodes:"                   $NumberOfNodes | tee -a $TMPDIR/nodes/summary
+  techo "etcd:"                        $NumberOfEtcdNodes | tee -a $TMPDIR/nodes/summary
+  techo "controlplane:"                $NumberOfControlplaneNodes | tee -a $TMPDIR/nodes/summary
+  techo "worker:"                      $NumberOfWorkerNodes | tee -a $TMPDIR/nodes/summary
+  techo "master:"                      $NumberOfMasterNodes | tee -a $TMPDIR/nodes/summary
+  techo "Pods Per Node"
+  for NODE in `${KUBECTL_CMD} get node -o NAME | awk -F '/' '{print $2}'`
+  do
+    NumberOfPodsPerNode=`${KUBECTL_CMD} get pods --all-namespaces --field-selector spec.nodeName=$NODE -o wide | wc -l`
+    techo $NODE" "$NumberOfPodsPerNode | tee -a $TMPDIR/nodes/pods-per-node
+  done
+}
+get-debug-tool-image() {
+  if [[ ! -z $IMAGE_FLAG ]]
+  then
+    DebugToolImage=$IMAGE_FLAG
+  else
+    DebugToolImage="leodotcloud/swiss-army-knife:latest"
+  fi
+  echo $DebugToolImage
+}
+deploy-serviceaccount() {
+  techo "Deploying serviceaccount"
+  ${KUBECTL_CMD} -n kube-system create serviceaccount cluster-health-check > /dev/null 2>&1
+  techo "Deploying clusterrole"
+  cat <<EOF | ${KUBECTL_CMD} -n kube-system apply -f -
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: cluster-health-check
+rules:
+- apiGroups:
+  - '*'
+  resources:
+  - '*'
+  verbs:
+  - '*'
+- nonResourceURLs:
+  - '*'
+  verbs:
+  - '*'
+EOF
+  techo "Deploying clusterrolebinding"
+  ${KUBECTL_CMD} create clusterrolebinding cluster-health-check \
+  --clusterrole=cluster-health-check \
+  --serviceaccount=kube-system:cluster-health-check > /dev/null 2>&1
+}
+cleanup-serviceaccount() {
+  techo "Deleting clusterrolebinding"
+  ${KUBECTL_CMD} delete clusterrolebinding cluster-health-check
+  techo "Deleting clusterrole"
+  ${KUBECTL_CMD} delete clusterrole cluster-health-check
+  techo "Deleting serviceaccount"
+  ${KUBECTL_CMD} -n kube-system delete serviceaccount cluster-health-check
+}
+deploy-swiss-army-knife() {
+  techo "Deploying swiss-army-knife test containers"
+  get-debug-tool-image
+  decho "DebugToolImage: $DebugToolImage"
```

Comment:
A good programming practice is to stick to a single naming convention. Either ALLCAPS or CamelCase or snakeCase. In general, all variables in shell scripts should be ALLCAPS.